src/autora/workflow/state/param.py

```diff
@@ -29,24 +30,46 @@ def _get_state_dependent_properties(state: SupportsControllerState):
         Nevertheless, we can iterate through its keys no problem:
         >>> [key for key in state_dependent_properties.keys()] # doctest: +NORMALIZE_WHITESPACE
         ['%observations.ivs[-1]%', '%observations.dvs[-1]%', '%observations.ivs%',
-        '%observations.dvs%', '%models[-1]%', '%models%']
+         '%observations.dvs%', '%experiment_data.conditions[-1]%',
+         '%experiment_data.observations[-1]%', '%experiment_data.conditions%',
+         '%experiment_data.observations%', '%models[-1]%', '%models%']
 
     """
 
     n_ivs = len(state.variables.independent_variables)
     n_dvs = len(state.variables.dependent_variables)
     state_dependent_property_dict = LazyDict(
         {
-            "%observations.ivs[-1]%": lambda: np.array(state.observations[-1])[
-                :, 0:n_ivs
-            ],
-            "%observations.dvs[-1]%": lambda: np.array(state.observations[-1])[
-                :, n_ivs:
-            ],
-            "%observations.ivs%": lambda: np.row_stack(
+            "%observations.ivs[-1]%": deprecate(
+                lambda: np.array(state.observations[-1])[:, 0:n_ivs],
+                "%observations.ivs[-1]% is deprecated, use %data.conditions[-1]% instead.",
+            ),
+            "%observations.dvs[-1]%": deprecate(
+                lambda: np.array(state.observations[-1])[:, n_ivs:],
+                "%observations.dvs[-1]% is deprecated, use %data.observations[-1]% instead.",
+            ),
+            "%observations.ivs%": deprecate(
+                lambda: np.row_stack(
+                    [np.empty([0, n_ivs + n_dvs])] + list(state.observations)
+                )[:, 0:n_ivs],
+                "%observations.ivs% is deprecated, use %data.conditions% instead.",
+            ),
+            "%observations.dvs%": deprecate(
+                lambda: np.row_stack(state.observations)[:, n_ivs:],
+                "%observations.dvs% is deprecated, " "use %data.observations% instead",
+            ),
+            "%experiment_data.conditions[-1]%": lambda: np.array(
+                state.observations[-1]
```

Comment:
this is an overdue attempt to clear up my confusion, but why are both the last observed independent (this line) and dependent (line 65) variables obtained by `state.observations[-1]`?

Reply:
That's because the `state.observations` property is an array where the first few columns are x's/independent variables, and then the last columns are y's/dependent variables. The underlying objects use the outdated naming convention, so they don't quite match.